pkg/assembler/backends/arangodb/certifyLegal.go

```diff
@@ -0,0 +1,882 @@
+  'timeScanned': certifyLegal.timeScanned,
+  'collector': certifyLegal.collector,
+  'origin': certifyLegal.origin
+}`)
+
+	cursor, err := executeQueryWithRetry(ctx, c.db, aqb.string(), values, "CertifyLegal")
+	if err != nil {
+		return nil, fmt.Errorf("failed to query for CertifyLegal: %w", err)
+	}
+	defer cursor.Close()
+
+	return c.getCertifyLegalFromCursor(ctx, cursor, decFilter, disFilter)
+}
+
+func getPkgCertifyLegalForQuery(ctx context.Context, c *arangoClient,
+	aqb *arangoQueryBuilder, values map[string]any,
+	decFilter, disFilter []*model.LicenseSpec) ([]*model.CertifyLegal, error) {
+	aqb.query.WriteString("\n")
+	aqb.query.WriteString(`RETURN {
+  'pkgVersion': {
+    'type_id': pType._id,
+    'type': pType.type,
+    'namespace_id': pNs._id,
+    'namespace': pNs.namespace,
+    'name_id': pName._id,
+    'name': pName.name,
+    'version_id': pVersion._id,
+    'version': pVersion.version,
+    'subpath': pVersion.subpath,
+    'qualifier_list': pVersion.qualifier_list
+  },
+  'certifyLegal_id': certifyLegal._id,
+  'declaredLicense': certifyLegal.declaredLicense,
+  'declaredLicenses': certifyLegal.declaredLicenses,
+  'discoveredLicense': certifyLegal.discoveredLicense,
+  'discoveredLicenses': certifyLegal.discoveredLicenses,
+  'attribution': certifyLegal.attribution,
+  'justification': certifyLegal.justification,
+  'timeScanned': certifyLegal.timeScanned,
+  'collector': certifyLegal.collector,
+  'origin': certifyLegal.origin
+}`)
+
+	cursor, err := executeQueryWithRetry(ctx, c.db, aqb.string(), values, "CertifyLegal")
+	if err != nil {
+		return nil, fmt.Errorf("failed to query for CertifyLegal: %w", err)
+	}
+	defer cursor.Close()
+
+	return c.getCertifyLegalFromCursor(ctx, cursor, decFilter, disFilter)
+}
+
+func setCertifyLegalMatchValues(aqb *arangoQueryBuilder, certifyLegalSpec *model.CertifyLegalSpec, queryValues map[string]any) {
+	if certifyLegalSpec.ID != nil {
+		aqb.filter("certifyLegal", "_id", "==", "@id")
+		queryValues["id"] = *certifyLegalSpec.ID
+	}
+	if certifyLegalSpec.DeclaredLicense != nil {
+		aqb.filter("certifyLegal", "declaredLicense", "==", "@declaredLicense")
+		queryValues["declaredLicense"] = *certifyLegalSpec.DeclaredLicense
+	}
+	if certifyLegalSpec.DiscoveredLicense != nil {
+		aqb.filter("certifyLegal", "discoveredLicense", "==", "@discoveredLicense")
+		queryValues["discoveredLicense"] = *certifyLegalSpec.DiscoveredLicense
+	}
+	if certifyLegalSpec.Attribution != nil {
+		aqb.filter("certifyLegal", "attribution", "==", "@attribution")
+		queryValues["attribution"] = *certifyLegalSpec.Attribution
+	}
+	if certifyLegalSpec.Justification != nil {
+		aqb.filter("certifyLegal", justification, "==", "@"+justification)
+		queryValues[justification] = *certifyLegalSpec.Justification
+	}
+	if certifyLegalSpec.TimeScanned != nil {
+		aqb.filter("certifyLegal", "timeScanned", "==", "@timeScanned")
+		queryValues["timeScanned"] = *certifyLegalSpec.TimeScanned
```

Comment:
store in UTC? I am updating the others verbs where timestamp is used also

Reply:
Done.